todo/views.py

```diff
@@ -0,0 +1,75 @@
+from django.shortcuts import get_object_or_404, redirect
+from django.urls import reverse_lazy
+from django.views import generic, View
+
+from todo.forms import TagForm, TaskForm
+from todo.models import Task, Tag
+
+
+class TaskListView(generic.ListView):
+    model = Task
+    context_object_name = "task_list"
+    template_name = "todo/task_list.html"
+    paginate_by = 3
+    queryset = Task.objects.prefetch_related("tags")
```

Comment:
According to the task requirements, tasks should be ordered from not done to done and from newest to oldest.
```
    def get_queryset(self):
        # Get queryset of tasks
        queryset = super().get_queryset()
        
        # Ensure tasks are ordered correctly: not done to done, and from newest to oldest
        queryset = queryset.order_by('is_done', '-created_at')
        
        return queryset
```
We call the superclass's get_queryset method to get the base queryset, which already includes all tasks.